src/openssl/ssl/context.cr

```diff
@@ -19,6 +19,63 @@ class OpenSSL::SSL::Context
     LibSSL.ssl_ctx_use_privatekey_file(@handle, file_path, LibSSL::SSLFileType::PEM)
   end
 
+  def ciphers=(ciphers : String)
+    LibSSL.ssl_ctx_set_cipher_list(@handle, ciphers)
+  end
+
+  def set_tmp_ecdh_key(curve = LibSSL::NID_X9_62_prime256v1)
+    key = LibSSL.ec_key_new_by_curve_name(curve)
+    LibSSL.ssl_ctx_ctrl(@handle, LibSSL::SSL_CTRL_SET_TMP_ECDH, 0, key)
+    LibSSL.ec_key_free(key)
+  end
+
+  def mode
+    LibSSL.ssl_ctx_ctrl(@handle, LibSSL::SSL_CTRL_MODE, 0, nil)
+  end
+
+  def mode=(mode)
+    LibSSL.ssl_ctx_ctrl(@handle, LibSSL::SSL_CTRL_MODE, mode, nil)
+  end
+
+  def clear_mode(mode = 0)
+    LibSSL.ssl_ctx_ctrl(@handle, LibSSL::SSL_CTRL_CLEAR_MODE, mode, nil)
+  end
+
+  def options
+    LibSSL.ssl_ctx_ctrl(@handle, LibSSL::SSL_CTRL_OPTIONS, 0, nil)
+  end
+
+  def options=(options)
+    LibSSL.ssl_ctx_ctrl(@handle, LibSSL::SSL_CTRL_OPTIONS, options, nil)
+  end
+
+  def clear_options(options = 0)
+    LibSSL.ssl_ctx_ctrl(@handle, LibSSL::SSL_CTRL_CLEAR_OPTIONS, options, nil)
```

Comment:
+1 for the above except `add_options` is unnecessary.  Treat it like any other enum.  `ctx.options |= LibSSL::SSL_OP_NO_SSLv2`.